src/pointer-analysis/value_set_domain.h

```diff
@@ -12,64 +12,125 @@ Author: Daniel Kroening, kroening@kroening.com
 #ifndef CPROVER_POINTER_ANALYSIS_VALUE_SET_DOMAIN_H
 #define CPROVER_POINTER_ANALYSIS_VALUE_SET_DOMAIN_H
 
-#define USE_DEPRECATED_STATIC_ANALYSIS_H
-#include <analyses/static_analysis.h>
+#include <analyses/ai.h>
+
+#include <util/invariant.h>
 
 #include "value_set.h"
 
 /// This is the domain for a value set analysis. It is intended to be the
 /// template parameter for `value_set_analysis_templatet`, so `VST` would be
 /// `value_sett`.
-template<class VST>
-class value_set_domain_templatet:public domain_baset
+template <class VST>
+class value_set_domain_templatet : public ai_domain_baset
 {
+protected:
+  /// ait checks whether domains are bottom to increase speed and accuracy.
+  /// Older frameworks don't so it is necessary to track this.
+  bool reachable;
+
 public:
   VST value_set;
 
-  // overloading
+  explicit value_set_domain_templatet(locationt l) : reachable(false)
+  {
+    value_set.clear();
+    value_set.location_number = l->location_number;
+  }
 
-  bool merge(const value_set_domain_templatet<VST> &other, locationt)
+  void make_bottom() override
   {
-    return value_set.make_union(other.value_set);
+    reachable = false;
+    value_set.clear(); //???
```

Comment:
?

Reply:
The underlying `value_sett` doesn't really have a clear idea of `top` and `bottom` because it's not really used / formalised like that.  The constructor of `value_set_domain_templatet` would always `value_set.clear()` which means I think it is the right thing to set when `make_bottom`.

Reply:
Perhaps replace the question marks by exactly that explanation?

Reply:
Done.